src/main/java/com/vmware/vipclient/i18n/messages/api/opt/local/LocalFileSystem.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright 2019-2020 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package com.vmware.vipclient.i18n.messages.api.opt.local;
+
+public class LocalFileSystem {
```

Comment:
This is used as a lock in LocalProductOpt. FileSystems.newFilesystem is not thread safe. This is evident when I put getSupportedLocales (refresh) in a separate thread. FileSystemAlreadyExistsException exception is thrown if another thread has opened the Opened the file system, making the current thread fail. We do not want the current thread to fail. Instead, we want it to wait for the other thread to finish so it could access the file system properly.

Reply:
I have changed this and moved the handling of thread safety in ProductService. Please check synchronized blocks in ProductService.

Reply:
Because we just read the file so lock is unnecessary.

Reply:
The file system is corresponding to uri, so it shouldn't to have only one lock for all file systems.

Reply:
> Because we just read the file so lock is unnecessary.

Try it with multiple threads and see the exception I said.

Reply:
> The file system is corresponding to uri, so it shouldn't to have only one lock for all file systems.

I have moved the lock to inside LocalProductOpt.

Reply:
Let's narrow down the scope to deliver this faster. Since this fix is not related to the Atlas bug, I have removed it from this PR and created this issue: #910.